### ISSUE TITLE
Could not load Ndless 4.5.3 in NSpire CX

### ISSUE THREAD

**Issue by Jojomonster175** [ndless-nspire/Ndless#240]
I have just download the new OS but the new version of Ndless shows (this is a deco) after I installed, how should I fix the problem?

**Comment by adriweb** [comment]
What version is shown in the about menu of the calc?
If it's not 4.5.3 it's not going to work. 
You need to download 4.5.3 specifically which doesn't show up on TI's website directly (they still show 4.5.2 for now)

**Comment by Jojomonster175** [comment]
I have download the new version 4.5.3.14 (shows on my calc)
Its seems like Ndless 4.5.3 aren't working at all
its just automaticallly restart after I download it(the new Ndless) without any NDLESS iNSTALLED showed on my calc

**Comment by adriweb** [comment]
Well that's a bug we can't seem to reproduce. As we can imagine it's been tested quite a few times and on several calculators. 

What's your calculator datecode/hardware revision? (The few characters that look like "N-1234AC" on the back, to the right of the serial number. ) When did you buy it?

**Comment by Jojomonster175** [comment]
> Well that's a bug we can't seem to reproduce. As we can imagine it's been tested quite a few times and on several calculators.
> 
> What's your calculator datecode/hardware revision? (The few characters that look like "N-1234AC" on the back, to the right of the serial number. ) When did you buy it?

P-0717AB

**Comment by Jojomonster175** [comment]
The calc is given from my friend so... not sure when it is bought from him

**Comment by ParkytGaming** [comment]
Same here on my CX CAS I have 4.5.3.14 installed and it doesn't load or install it at all it just sits on the Ndless screen and does nothing. It did reboot the first couple of attempts though.

**Comment by Vogtinator** [comment]
Did you transfer `ndless_resources.tns` correctly?

**Comment by ParkytGaming** [comment]
> Did you transfer `ndless_resources.tns` correctly?

Is there a correct way to do that?

**Comment by ParkytGaming** [comment]
> Did you transfer `ndless_resources.tns` correctly?

Is there a correct way to do that?
It was transferred, and they were in the same folder too🤔...

**Comment by ParkytGaming** [comment]
It just doesn't do anything. Ndless has worked previously on my Nspire. However, until I mistakenly updated my Nspire, this new version hasn't worked... I can verify I am running OS version 4.5.3

**Comment by Danke-Boi** [comment]
I’m no expert, but have you tried turning it off then back on again? XD In all seriousness, do you have files you want to keep on it? Maybe you could factory reset, then update again to the newest version? Again, I’m no expert, so I would try EVERYTHING else first, but maybe it would work?

**Comment by Danke-Boi** [comment]
Or, something mine did is, ndless installed, but it didn’t show up, try running a ndless app. If it works, Yay! I’m if not then ask somebody who knows what they are doing.

**Comment by ParkytGaming** [comment]
Isn't the installer the app? There isn't a separate app tho as far as I know. Unless that's the Ndless_Resources and I have to run that. But since it's a .tns file my Nspire won't let it run without Ndless installed.

**Comment by adriweb** [comment]
They were saying, in order to check that ndless was installed, run any app (or the ndless_resources, since that will ask you if you want to uninstall it (by resetting), if ndless is indeed installed).

**Comment by MLGityaJtotheA** [comment]
I can confirm that I was in the same situation but after copying ndless_resources.tns over as well, it immediately worked.

**Comment by Jojomonster175** [comment]
> They were saying, in order to check that ndless was installed, run any app (or the ndless_resources, since that will ask you if you want to uninstall it (by resetting), if ndless is indeed installed).

I tried and it failed :( 
Are there any other ways to solve this prob.

**Comment by Danke-Boi** [comment]
In your first message, you said you updated your calculator, but did you
get the appropriate Ndless installer? Even if you did, try deleting it and
putting it back on the calc.

**Comment by Danke-Boi** [comment]
If all of this fails, you can down-date (I think that’s the opposite of
update) to the previous OS. There is a tutorial here
https://m.youtube.com/watch?v=ZNBxjZNI10s
Sorry I don’t know how to link things on mobile.  If this doesn't work. I’m
sure there is a tutorial on tiplanet.org or whatever it is called.

**Comment by lights0123** [comment]
Downgrading support was removed years ago. Current techniques require Ndless anyways, and probably don't support OS 4.5.3 yet anyways.

**Comment by Danke-Boi** [comment]
Oh, well would a factory reset downgrade the os?

**Comment by lights0123** [comment]
Nope. You were able to [flash an old bootloader](https://www.reddit.com/r/nspire/comments/466kkc/help/) with the serial port on the dock connector on the bottom of the calculator, but I have no clue if that still works.

**Comment by unilock** [comment]
I'm having a similar (if not identical) issue. I have a TI-Nspire CX (P-0615V) on 4.5.3.14, notably with nBoot and ControlX 5.0.2 installed.
Typically ControlX would automatically install Ndless, but it seems this has been disabled on OS versions above 4.5.1 -- or, more likely, it's not working.
When I run ndless_installer_4.5.3, the installer screen (with "this line is just deco") shows up for a few seconds, then the calculator reboots. After rebooting, there is no "Ndless installed!" text apparent, nor can I open any native programs on the calculator (most notably ndless_resources, which responds "This document format is not supported").

**Comment by Vogtinator** [comment]
ControlX ships its own installer, which needs to be updated for every new OS version. The ndless installer doesn't trigger or need a reboot, it means that it crashed. I assume you tried this multiple times?

**Comment by unilock** [comment]
That's neat, that ControlX ships its own installer. Guess I'll have to wait for it to be updated in order to have Ndless auto-installed. Unfortunately, I can't update it without Ndless...
I have indeed tried running the installer multiple times. I've also tried re-copying ndless_resources and ndless_installer_4.5.3 to the calculator, so I know they're valid -- at least, they're the same as in the Ndless r2016 release archive.

**Comment by Vogtinator** [comment]
Does the behaviour change if you delete or rename `ndless_resources`?

**Comment by unilock** [comment]
Yes. If I rename `ndless_resources`, the installer opens and the cursor appears. After a few seconds, the cursor disappears. Nothing happens after that.

**Comment by Vogtinator** [comment]
Ok, that means it most likely crashes inside `ndless_resources`, which is strange.

**Comment by unilock** [comment]
Ndless installs correctly with ControlX disabled (by booting while holding `esc`+`menu`+`-`). Now, a different issue: I have Nover and nSonic2MS set to run at startup. I can't tell whether Nover is working, but nSonic2MS doesn't appear to be; the application doesn't come up when viewing My Documents, and when running it manually, it reports "Hook already installed".
Also, the "Ndless installed!" text disappears after a while. I'm not sure whether this is intentional.

**Comment by pagkly** [comment]
So does this issue still persisted? I have not upgrade to 4.5.3 yet. Im still at 4.5.1

**Comment by RuralAnemone** [comment]
This issue will always happen if your version of Ndless does not match your calculator's OS version
to see your calc's OS version, do the following

1. Press Home/on
2. Press 5
3. Press 4

(this goes from home to settings to status)

**Comment by RuralAnemone** [comment]
> So does this issue still persisted? I have not upgrade to 4.5.3 yet. Im still at 4.5.1

This issue will always happen if your version of Ndless does not match your calculator's OS version
to see your calc's OS version, do the following

1. Press Home/on
2. Press 5
3. Press 4

(this goes from home to settings to status)

**Comment by RuralAnemone** [comment]
also, You will need to transfer ndless_resources.tns for this to work

**Comment by unilock** [comment]
@RuralAnemone I assume @pagkly is asking whether the issue still exists on 4.5.3, before updating to said version from 4.5.1. It does, of course, since there have been no Ndless releases since.

**Comment by RuralAnemone** [comment]
> @RuralAnemone I assume @pagkly is asking whether the issue still exists on 4.5.3, before updating to said version from 4.5.1. It does, of course, since there have been no Ndless releases since.

I'm just saying this for general reference

**Comment by nichales** [comment]
new problem for ndless there 4.5.3 to 4.5.4 ndless and new cx ii 5.2 to 5.3 ndless both not working
first cx 4.5.3 to 4.5.4 both in ndless all went to shut down to restart ti-nspire that why i know what happen and cx ii 5.2 to 5.3 i opened file ndless then loading too longest time like 5 to 6 hour waste battery i report that ndless you need fix correct that problem!

**Comment by nichales** [comment]
one thing when you update that ti-nspire cx or ti-nspire cx ii you can't charge that os to downgrade until new os and new ndless
like more 4 years! or more that to wait!

**Comment by Facts-and-Logic** [comment]
For the CX II on 5.3, it works for me. Sometimes it hangs at the install screen so I just press the reset button at the back of the calc and reinstall it. It seems like creating a new document or opening another document before installing ndless causes it to hang during the installation.

**Comment by Danke-Boi** [comment]
I have similar issues, and you have to make sure that in the ndless folder
you have the ndless_resources file as well as the ndless installer. Also,
sometimes the program crashes, you just have to reset the calculator  using
a pencil or pen or paper clip and pressing the button on the back labeled
“reset”. It reboots the calculator manually, IT DOES NOT FACTORY RESET THE
CALCULATOR.

**Comment by nichales** [comment]
CX II had problem its loading long time when you reset fresh to show up then folder to ndless will loadint too long time use reset to fresh by whatever i tell you you cant mix same on 4.5.3 with 4.5.4 or 5.2 with 5.3 will mess up i tell you easy better use one target like 5.3 will keep target not two will mess up it.

**Comment by nichales** [comment]
nope its not working on ti-nspire cx ii it too long loading until my battery died that i don't think so not help that answer.
:(

**Comment by nichales** [comment]
aaer you read? this i wait for long time for fix ndless to working gameboy on CX II do not use same like a 5.2 with 5.3 will mess up for install ndless one time each time.
